### ISSUE TITLE
Pinball loss for quantile predictions

### ISSUE THREAD

**Issue by trdvangraft** [fastai/fastai#3279]
**Is your feature request related to a problem? Please describe.**

For a regression problem, you are often not interested in the actual outcome but more in the boundary where the outcomes lay between.  This can be achieved through for example taking a quantile output. This however often requires taking the quantile loss function (Pinball loss), which is not natively supported within Fastai. I think that fastai can benefit from this loss function or at the very least give more flexibility to the user.

**Describe the solution you'd like**
I would like to implement the quantile loss function similarly how it is done in the following [post](https://www.evergreeninnovations.co/blog-quantile-loss-function-for-machine-learning/). 
TLDR:  `def quantile_loss(q, y_p, y): e = y_p-y
        return tf.keras.backend.mean(tf.keras.backend.maximum(q*e, (q-1)*e))`

**Describe alternatives you've considered**
Afaik there are no alternatives to the quantile loss function as suggested in this issue.

**Additional context**
I am completely able to implement this function myself, however I thought it would be nice to test if there was any interest into this function

**Comment by Sandy4321** [comment]
great idea 
thanks
when it will be ready ?

**Comment by Sandy4321** [comment]
it is not implemented yet
https://docs.fast.ai/losses.html#DiceLoss
class BaseLoss
class CrossEntropyLossFlat
class FocalLoss
class FocalLossFlat
class BCEWithLogitsLossFlat
BCELossFlat
MSELossFlat
L1LossFlat
class LabelSmoothingCrossEntropy
class LabelSmoothingCrossEntropyFlat
class DiceLoss

**Comment by ThomasBourgeois** [comment]
That would clearly be a great add-on.

**Comment by KossaiSbai** [comment]
If people are still interested in having that function implemented, more than happy to take this on
